libflux/src/core/parser/mod.rs

```diff
@@ -577,6 +577,76 @@ impl Parser {
             MonoType::Invalid
         }
     }
+
+    // Record = "{" [ Identifier (Suffix1 | Suffix2) ] "}"
+    // Suffix1 = ":" MonoType { "," Property }
+    // Suffix2 = "with" [Properties]
+    #[cfg(test)]
+    fn parse_record(&mut self) -> Record {
+        let start = self.open(TOK_LBRACE, TOK_RBRACE);
+        let mut properties: Option<Vec<PropertyType>> = None;
+        let property: PropertyType;
+
+        let mut id: Option<Identifier> = None;
+        let _id;
+        let t = self.peek();
+        if t.tok == TOK_IDENT {
+            // Indentifier
+            _id = self.parse_identifier();
+            // suffix one needs attention
+            let t2 = self.peek();
+            if t2.tok == TOK_COLON {
+                let mut ps = Vec::<PropertyType>::new();
+                self.consume(); // :
+                let mt = self.parse_monotype();
+                property = PropertyType {
+                    base: base_from_monotype(&mt),
+                    identifier: _id,
+                    monotype: mt,
+                };
+                ps.push(property);
+                while self.peek().tok == TOK_COMMA {
+                    self.consume(); // ,
+                    ps.push(self.parse_property());
+                }
+                properties = Some(ps);
+            } else if t2.lit == "with" {
+                self.consume(); // consume the with
+                properties = self.parse_properties();
+                id = Some(_id);
+            }
+        }
+
+        let end = self.close(TOK_RBRACE);
+
+        Record {
+            base: self.base_node_from_tokens(&start, &end),
+            tvar: id,
+            properties,
+        }
+    }
+    #[cfg(test)]
+    fn parse_properties(&mut self) -> Option<Vec<PropertyType>> {
+        let mut properties = Vec::<PropertyType>::new();
+        properties.push(self.parse_property());
+        // check for more properties
+        while self.peek().tok == TOK_COMMA {
+            self.consume();
+            properties.push(self.parse_property());
+        }
+        return Some(properties);
+    }
+    #[cfg(test)]
+    fn parse_property(&mut self) -> PropertyType {
+        let identifier = self.parse_identifier(); // identifier
+        self.expect(TOK_COLON); // :
+        let monotype = self.parse_monotype();
+        PropertyType {
+            base: base_from_monotype(&monotype),
```

Comment:
Can you change this to `self.base_node_from_others(&identifier.base, &monotype.base)`?

Reply:
👍